src/System.Globalization/tests/CharNumericCategory/GetNumericValue.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.Globalization;
+using UnicodeHelpers;
+
+namespace System.Globalization.Tests
+{
+    public class GetNumericValue
+    {
+        [Fact]
+        public void NullCheck()
+        {
+            Assert.Throws<ArgumentNullException>(() => CharUnicodeInfo.GetNumericValue(null, 0));
+        }
+
+        [Fact]
+        public void InputCharWithValue()
+        {
+            TestCharInput(true);
+        }
+
+        [Fact]
+        public void InputCharWithoutValue()
+        {
+            TestCharInput(false);
+        }
+
+        [Fact]
+        public void InputStringWithoutValue()
+        {
+            TestStringInputWithIndex(String.Empty, String.Empty, false);
+        }
+
+        [Fact]
+        public void InputStringWithValue()
+        {
+            TestStringInputWithIndex(String.Empty, String.Empty, true);
+        }
+
+        [Fact]
+        public void InputStringMiddleOfStringWithValue()
+        {
+            TestStringInputWithIndex("some text", "more text", true);
+        }
+
+        [Fact]
+        public void InputStringMiddleOfStringWithoutValue()
+        {
+            TestStringInputWithIndex("some text", "more text", false);
+        }
+
+        [Fact]
+        public void InputStringEndOfStringWithValue()
+        {
+            TestStringInputWithIndex("some text", String.Empty, true);
+        }
+
+        [Fact]
+        public void InputStringEndOfStringWithoutValue()
+        {
+            TestStringInputWithIndex("some text", String.Empty, false);
+        }
+
+        [Fact]
+        public void InputStringBeginningOfStringWithValue()
+        {
+            TestStringInputWithIndex(String.Empty, "ending text", true);
+        }
+
+        [Fact]
+        public void InputStringBeginningOfStringWithoutValue()
+        {
+            TestStringInputWithIndex(String.Empty, "ending text", false);
+        }
```

Comment:
FYI for cases like this, all of these different facts can be condensed into a single theory, e.g.
```C#
[Theory]
[InlineData("", "", false)]
[InlineData("", "", true)]
[InlineData("some text", "more text", false)]
[InlineData("some text", "", true)]
[InlineData("some text", "", false)]
[InlineData("", "ending text", true)]
[InlineData("", "ending text", false)]
public void TestStringInputWithIndex(string frontPadding, string backPadding, bool hasValue)
{
    ...
}
```
This keeps the tests more concise, makes it easy to add additional data inputs as needed, keeps all of the related tests together, and helps to highlight the differences between test cases.